File: host-v2/fixtures/builtin_stubs_extended.mjs
// Tier-Ω.5.s: smoke fixture for the five new node-builtin stubs.
import a from "node:assert";
import https from "node:https";
import s from "node:stream";
import url from "node:url";
import util from "node:util";

// node:assert — ok(true) returns undefined.
a.ok(true);
console.log("assert.ok(true) ok");

// node:https — request is a function (stub, but typeof check works).
console.log("https.request typeof:", typeof https.request);

// node:stream — Readable is a function.
console.log("stream.Readable typeof:", typeof s.Readable);

// node:url — fileURLToPath round-trips.
console.log("fileURLToPath:", url.fileURLToPath("file:///tmp/x"));

// node:util — format does %s substitution.
console.log("util.format:", util.format("hi %s", "world"));

File: host-v2/tests/builtin_stubs_extended.rs
//! Tier-Ω.5.s: smoke tests for the assert / https / stream / url / util
//! built-in stubs. Drives the host-v2 binary against
//! fixtures/builtin_stubs_extended.mjs and asserts the expected stdout
//! lattice.

use std::process::Command;

fn bin() -> &'static str {
    env!("CARGO_BIN_EXE_rusty-bun-host-v2")
}

fn run() -> std::process::Output {
    let fixture = concat!(env!("CARGO_MANIFEST_DIR"), "/fixtures/builtin_stubs_extended.mjs");
    Command::new(bin()).arg(fixture).output().expect("run host-v2")
}

#[test]
fn t01_assert_ok_true_succeeds() {
    let out = run();
    assert!(out.status.success(), "binary failed: {:?}", out);
    let stdout = String::from_utf8_lossy(&out.stdout);
    assert!(stdout.contains("assert.ok(true) ok"), "stdout: {stdout}");
}

#[test]
fn t02_https_request_typeof_function() {
    let out = run();
    assert!(out.status.success(), "binary failed: {:?}", out);
    let stdout = String::from_utf8_lossy(&out.stdout);
    assert!(stdout.contains("https.request typeof: function"), "stdout: {stdout}");
}

#[test]
fn t03_stream_readable_typeof_function() {
    let out = run();
    assert!(out.status.success(), "binary failed: {:?}", out);
    let stdout = String::from_utf8_lossy(&out.stdout);
    assert!(stdout.contains("stream.Readable typeof: function"), "stdout: {stdout}");
}

#[test]
fn t04_url_file_url_to_path() {
    let out = run();
    assert!(out.status.success(), "binary failed: {:?}", out);
    let stdout = String::from_utf8_lossy(&out.stdout);
    assert!(stdout.contains("fileURLToPath: /tmp/x"), "stdout: {stdout}");
}

#[test]
fn t05_util_format_substitution() {
    let out = run();
    assert!(out.status.success(), "binary failed: {:?}", out);
    let stdout = String::from_utf8_lossy(&out.stdout);
    assert!(stdout.contains("util.format: hi world"), "stdout: {stdout}");
}
